ambry-store/src/test/java/com.github.ambry.store/BlobStoreStatsTest.java

```diff
@@ -75,6 +99,35 @@ public BlobStoreStatsTest(boolean isLogSegmented) throws InterruptedException, I
   public void cleanup() throws InterruptedException, IOException, StoreException {
     state.destroy();
     assertTrue(tempDir.getAbsolutePath() + " could not be deleted", StoreTestUtils.cleanDirectory(tempDir, true));
+    indexScannerScheduler.shutdown();
+    if (!indexScannerScheduler.awaitTermination(1000, TimeUnit.MILLISECONDS)) {
+      fail("Could not terminate index scanner task after BlobStoreStats shutdown");
+    }
+    queueProcessorScheduler.shutdown();
+    if (!queueProcessorScheduler.awaitTermination(1000, TimeUnit.MILLISECONDS)) {
+      fail("Could not terminate queue processor task after BlobStoreStats shutdown");
+    }
+  }
+
+  /**
+   * Test {@link BlobStoreStats} can be initialized (with bucketing enabled/disabled) and closed properly.
+   * @throws InterruptedException
+   */
+  @Test
+  public void testInitializationAndClose() throws InterruptedException {
+    int bucketCount = bucketingEnabled ? 1 : 0;
+    final CountDownLatch scanStartedLatch = new CountDownLatch(1);
+    MockThrottler mockThrottler = new MockThrottler(scanStartedLatch, new CountDownLatch(0));
+    throttlers.put(BlobStoreStats.IO_SCHEDULER_JOB_TYPE, mockThrottler);
```

Comment:
you don't seem to reset the throttlers for every test. So, ensure that every test resets or adds this entry for interested job. if not, it might be using the one that was added from a previous test case.